### PR TITLE
Add list design doc

### PR DESCRIPTION
**Pull request by wyvl**
INT-14085

**The following MUST be checked prior to approval. If not relevant to your PR, remove the line from your description.**
- [ ]  The `styleguide.config.js` has been updated to show examples/new functionality for the component
- [ ]  Test Coverage for any new or updated functionality
- [ ]  New and updated components have been tested locally in lapis and SPA and work as expected
- [ ]  This PR has been tagged with PATCH, MINOR, or MAJOR.
- [ ]  The component has data-component-type and data-component-context attributes following the standard
- [ ]  You have requested a cross-team review on this component so everyone knows it exists

### CODE REVIEW

**Comment by wyvl** [comment]
Screen Shot 2021-01-05 at 4 13 50 pm
Screen Shot 2021-01-05 at 4 14 03 pm
Screen Shot 2021-01-05 at 4 14 12 pm
Screen Shot 2021-01-05 at 4 14 20 pm
Screen Shot 2021-01-05 at 4 14 27 pm
Screen Shot 2021-01-05 at 4 14 35 pm

**Review by andrewhillcode — COMMENTED** [review]
I'll finish this review tomorrow

File: docs/List.md [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+Pagination section would not have a skeleton for it
+
+```jsx
+import { BlockSkeleton } from '@Domain/Skeletons';
+import { Inline } from '@Domain/Layouts';
+import { Variables } from '@Common';
+
+<>
+  <BlockSkeleton margins={{ bottom: Variables.Spacing.sLarge}} height={Variables.Spacing.s2XLarge}/>
+  <Inline
```
Comment (andrewhillcode) [inline docs/List.md:452]: Should this also be in a carousel? not sure if that works with inline though.

File: docs/List.md [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+
+### Loading state
+List section (from top to bottom): filter and sort section, statistics tile, cards, pagination with sLarge spacing 24px in between
+
+Filter and sort section goes together as a 40px height skeleton
+Statistics tile would depend on the number of tile would the list have (It would be depend on minimum number of tile so it would not include the optional ones)
+The skeleton would be 200 x 90 with 8 margin in between
Cards section skeleton would have three cards only and the skeleton height would depend on the card height on that page
+
+Pagination section would not have a skeleton for it
```
Comment (andrewhillcode) [inline docs/List.md:443]:
```When loading content we display skeletons (not spinners). The skeletons should reflect the structure of the content that will be displayed once the page has loaded. 

The following values will typically work for most content, but depending on the size of your content they may need minor adjustments.

Filter Section: 40px height
Statistics: 200px wide | 90px height | sLarge spacing between 
Cards: 78px height | sMedium spacing between

No skeleton for pagination bar

Only display the minimum number of Statistics/Cards that the page can load with. If it always has 6 Statistics, then show 6 skeletons. For lists of Cards, display as many as a full page of results (typically 10) unless the content is unlikely to reach a full page of results.
```

File: docs/List.md [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+                    handleDatesChange={(dates) => { setState({ startDate: dates.startDate, endDate: dates.endDate })}}
+                  />}
+/>
+</>
+```
+<br />
+
+### Statistic section
+The label text length should be limited to one line with additional info in ( ) 
The value should be in one line too.
```
[the remark this answers is not in the document]
Comment (wyvl) [reply docs/List.md:332]: total Cost (AUD)
instead of total cost in AUD

File: docs/List.md [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+        />
+      </Carousel.Tile>
+      <Carousel.Tile>
+        <Statistic
+          title='Missing End Dates'
+          value='3'
+        />
+      </Carousel.Tile>
+    </Carousel>
+    <Carousel>
```
[the remark this answers is not in the document]
Comment (wyvl) [reply docs/List.md:361]: i just split them two examples as i want to show the order

File: docs/List.md (hunk 1 above)
Comment (wyvl) [reply docs/List.md:452]: inline is simpler and just want to have 8 px gap

File: docs/List.md (hunk 2 above)
Comment (wyvl) [reply docs/List.md:443]: should card skeleton depend on the content or fixed height?

File: docs/List.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (andrewhillcode) [reply docs/List.md:332]: Something like that is probably better off in our language guidelines 🤷

[the remark this answers is not in the document]
Comment (andrewhillcode) [reply docs/List.md:332]: also I'm talking about this square thing:
image

File: docs/List.md (hunk 2 above)
Comment (andrewhillcode) [reply docs/List.md:443]: Card skeleton should match the content height of how big the cards are going to be. 78px is the normal size right?

File: docs/List.md (hunk 1 above)
Comment (andrewhillcode) [reply docs/List.md:452]: i mean the inline within a carousel. or else mobile skeletons might look weird.

**Review by andrewhillcode — APPROVED** [review]

**Pull request merged**